packages/editor/src/components/rich-text/format-toolbar/index.js

```diff
@@ -164,8 +129,6 @@ class FormatToolbar extends Component {
 			rel: this.state.opensInNewWindow ? 'noreferrer noopener' : null,
 			value,
 		} } );
-
-		this.setState( { linkValue: value, isEditingLink: false } );
 		if ( ! this.props.formats.link.value ) {
```

Comment:
We call `this.props.onChange` so we don't need to update state explicitly because `getDerivedStateFromProps` will handle it properly.